src/Evolution/DiscontinuousGalerkin/ObserveFields.hpp

```diff
@@ -225,7 +240,17 @@ class ObserveFields<VolumeDim, tmpl::list<Tensors...>,
       }
     };
     EXPAND_PACK_LEFT_TO_RIGHT(record_errors(
-        tmpl::type_<AnalyticSolutionTensors>{}, analytic_solution_tensors));
+        tmpl::type_<AnalyticSolutionTensors>{},
+        local_analytic_solution_tensors));
+        },
+        [
+        ](std::false_type /*has_analytic_solution*/,
+          const auto& /*local_cache*/,
+          const db::item_type<
+              AnalyticSolutionTensors>&... /*local_analytic_solution_tensors*/)
+        noexcept {})(
+        detail::has_analytic_solution_alias<Metavariables>{}, cache,
```

Comment:
Overload on whether `AnalyticSolutionTensors` is empty instead of the contents of the metavariables.  In fact, I think passing the cache as a (type `auto`) third argument to `record_errors` and fetching the solution inside that function might work without the overloader.

Reply:
It took me a minute to figure out how that would work, but good idea!